### ISSUE TITLE
Reduction in Bundle Size/ Tree shaking problem?

### ISSUE THREAD

**Issue by deevo-sage** [tremorlabs/tremor-npm#605]
### What problem does this feature solve?

<img width="1303" alt="image" src="https://github.com/tremorlabs/tremor/assets/54073655/553605b0-f83a-4aff-90c0-b56525d47269">
<img width="1226" alt="image" src="https://github.com/tremorlabs/tremor/assets/54073655/93aa7afc-8b6f-4a07-b17b-d62829962260">

The bundle size increases drastically when using tremor. but if I importing directly from the Dist folder, if you see the '/real-user-report' path

<img width="1262" alt="image" src="https://github.com/tremorlabs/tremor/assets/54073655/97143e64-35ad-4556-8c0d-bc7948278206">

just importing from the dist path reduced the bundle size by 200 kb and I'm just using Card on that page

for the '/performance' path, I'm using a lot more components and the bundle size was 430KB and again just shifting to dist import for all the charts and elements the bundle size reduced around 200KB 

### What does the proposed API look like?

I Assume there is some issue with Tree Shaking.

**Comment by severinlandolt** [comment]
#493 #489

**Comment by motinados** [comment]
I tried a few things using Vite.

1. Vite [as per tutorial](https://www.tremor.so/docs/getting-started/installation)
![vite1-result](https://github.com/tremorlabs/tremor/assets/69296148/0f8d865d-59db-4e99-b231-4d0951b91bb8)

2. Vite + deep imports
![vite2-result](https://github.com/tremorlabs/tremor/assets/69296148/01588a90-8cea-4cf6-ab3b-1ac1182c692d)

3. Vite + '"sideEffects": false'
added '"sideEffects": false' to './node_modules/@tremor/react/package.json'
Please note that I haven't verified the functionality.
![vite3-result](https://github.com/tremorlabs/tremor/assets/69296148/5d45dfcf-f313-4f05-994c-b8a85772dd06)

The Code is [here](https://github.com/motinados/tremor-build-size-test)

Many libraries have a sideEffect field, but tremor does not. Wouldn't this be a solution?

**Comment by mitrotasios** [comment]
yes indeed, that's most likely the solution. Thanks for trying that out! Feel free to open a PR and add the toggle to the `package.json` if you like :)

**Comment by motinados** [comment]
I also tried a few things with Nextjs.

+ Nextjs13
![next1-result](https://github.com/tremorlabs/tremor/assets/69296148/9c881d84-01f9-4d53-8d44-bed89414a1d4)

+ Nextjs13 + deep imports
![next2-result](https://github.com/tremorlabs/tremor/assets/69296148/5640dae0-8ab7-41a0-9f69-434c281f3837)

+ Nextjs13 + sideEffects: false  
![next3-result](https://github.com/tremorlabs/tremor/assets/69296148/e404cc01-33ec-472c-b5b7-c2ead602ef7a)

Just the sideEffects field is not enough for Nextjs.

**Comment by mitrotasios** [comment]
Thank you!

**Comment by DostalJ** [comment]
I am curious. Are changes to the tremor source code feasible and planned, such that the bundle size with normal imports can be as small as it is with deep import currently?

Thank you very much for providing us with a great dashboarding solution Tremor has become.

**Comment by motinados** [comment]
This code works for my nextjs app. 
Can someone please try it?
Note that next@canary is required.

next.config.js
```javascript
/** @type {import('next').NextConfig} */
const nextConfig = {
    experimental: {
        optimizePackageImports: [
          "@tremor/react"
        ]
    }
}
module.exports = nextConfig
```

package.json
```
"next": "^13.4.20-canary.12"
```

**Comment by DostalJ** [comment]
Results of `optimizedPackegeImports` on my app with the latest
`"next": "^13.4.20-canary.12"`
Thanks for help!

Before:
<img width="603" alt="Screenshot 2023-09-11 at 15 29 13" src="https://github.com/tremorlabs/tremor/assets/20856372/0a4ef6c0-b68f-4c0e-adef-2c0e15a4b822">


After:
<img width="595" alt="Screenshot 2023-09-11 at 15 29 19" src="https://github.com/tremorlabs/tremor/assets/20856372/dc57111f-deb6-49d2-af58-12dd3a0d7cfb">

**Comment by angelhodar** [comment]
@DostalJ In my case the `optimizePackageImports` doesnt make a big impact because I am only using the charts from tremor, the rest of the components are from headlessui and shadcn. In your case I suppose you just import the tremor button for example in one page, and in that cases is where the impact is much bigger.

**Comment by paullaros** [comment]
Looks like Next.js is optimizing `@tremor/react` by default: https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports
